### PR TITLE
support new js sdk with oauth 2.0

### PR DESCRIPTION
**Pull request by ahilles107**
Hi,

This commit bring support to new cookie "fbsr_" and update SessionPeresistance acording Facebook solution.

it fully work with my prev PR.

Refactored. Please remove prev one.

### CODE REVIEW

File: Facebook/FacebookSessionPersistence.php [hunk 1]
```diff
@@ -59,7 +80,12 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function clearPersistentData($key)
     {
-        $this->session->remove($this->prefix.$key);
+        if (!in_array($key, self::$kSupportedKeys)) {
+            self::errorLog('Unsupported key passed to clearPersistentData.');
+	        return;
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php]: bad indentation :)

File: Facebook/FacebookSessionPersistence.php [hunk 2]
```diff
@@ -48,7 +59,17 @@ class FacebookSessionPersistence extends \BaseFacebook
      */
     protected function getPersistentData($key, $default = false)
     {
-        return $this->session->get($this->prefix.$key, $default);
+        if (!in_array($key, self::$kSupportedKeys)) {
+            self::errorLog('Unsupported key passed to getPersistentData.');
+            return $default;
+        }
+        $sessionVariableName = $this->constructSessionVariableName($key);
```
Comment (willdurand) [inline Facebook/FacebookSessionPersistence.php:67]: You should before a new line after this one.

(edited: before this line instead of after)

**Comment by ahilles107** [comment]
something more?

**Comment by marbemac** [comment]
Have you actually tested this? Is it working on your machine? Right off the bat I get a javascript error:

"OAuth2 specification states that 'perms' should now be called 'scope'. Please update."

You need to change the 'perms' parameter in the two twig templates to 'scope'.

Even after all of these updates the bundle does not seem to work. A die('foo'); in loadUserByUsername() in my provider script never gets called. It did work before the move to the 3.0 SDK, grr.

If I can help by testing anything you think could cause this I'm more than happy to, just let me know.

**Comment by lsmith77** [comment]
i tested it with https://github.com/ahilles107/symfony-facebook .. and it appeared to be working fine ..

**Comment by ahilles107** [comment]
Yes, i missed this. For loginButton perms must be changed to scope - i don't use js authentication. Server side flow works perfect for my apps. I will fix this tomorrow.

You heave own provider? Do You have fbsr_ cookie (instead fbs_ )? You have signed_request in Request?

**Comment by marbemac** [comment]
Hey, 

I have a fbsr_ in request, but no signed_request. Also, the FB login button stays as a login button no matter if I am logged into facebook or not.

**Comment by lsmith77** [comment]
would be great if you two could work out those issues .. i am sorry but the facebook connect app i created got moved back to the client, so i am lacking a use case for myself.

**Comment by marbemac** [comment]
After some investigating it seems like the current Javascript SDK might not play with with the PHP 3.x SDK. I'll continue to look into it and update if I find an answer.

In the meantime, ahilles do you have an easy way to quickly test using the javascript login/logout buttons? If it doesn't work for you then we pretty much know it's a javascript problem.

**Comment by ahilles107** [comment]
@marbemac you have 3.1.1 php-sdk, and do you have activated oauth2.0 in facebook app?

**Comment by marbemac** [comment]
That was it. Wow I feel silly now. I hadn't realized manually enabling oath2 was required. Thanks so much for your help!

**Comment by lsmith77** [comment]
@ahilles107: is this hint in the docs? what else are we missing now to have everything working?

**Comment by ahilles107** [comment]
@lsmith77 oauth is checked by default. Only old app's must do this. i just working about change permission to scope.

**Comment by logiclabz** [comment]
Facebook login connect has migrated to oauth2.0 creating Facebook Login Error as Oauth2 specification states that 'perms' should now be called 'scope'
http://blog.logiclabz.com/javascript/facebook-login-error-oauth2-specification-states-that-perms-should-now-be-called-scope.aspx

**Pull request merged**